smoke-test/src/test/java/org/opennms/smoketest/OpenNMSCloudPluginIT.java

```diff
@@ -0,0 +1,165 @@
+        install_cloud_plugin_in_sentinel_must_be_successful();
+    }
+
+    public void install_cloud_plugin_in_opennms_must_be_successful() throws Exception {
+        // Given
+        stack.opennms().copyFileToContainer(MountableFile.forHostPath(PATH_TO_FILE), format("%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        String config = createConfig();
+
+        // When
+        opennmsShell.runCommand(String.format("kar:install file:%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        opennmsShell.runCommand("feature:install opennms-plugin-cloud-core");
+        opennmsShell.runCommand("feature:list | grep opennms-plugin-cloud-core", output -> output.contains("Started"));
+        opennmsShell.runCommand(config);
+        opennmsShell.runCommand("opennms-cloud:init key", output -> output.contains("Initialization of cloud plugin in OPENNMS was successful."));
+    }
+
+
+    public void install_cloud_plugin_in_sentinel_must_be_successful() throws Exception {
+        // Given
+        stack.sentinel().copyFileToContainer(MountableFile.forHostPath(PATH_TO_FILE), format("%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        String config = createConfig();
+
+        // When
+        sentinelShell.runCommand(String.format("kar:install file:%s%s", CONTAINER_PATH, KAR_FILE_NAME));
+        sentinelShell.runCommand("feature:install opennms-plugin-cloud-sentinel");
+        sentinelShell.runCommand("feature:list | grep opennms-plugin-cloud-sentinel", output -> output.contains("Started"));
+        sentinelShell.runCommand(config);
+        sentinelShell.runCommand("opennms-cloud:init key", output -> output.contains("Initialization of cloud plugin in SENTINEL was successful."));
+    }
+
+    private String createConfig() throws IOException {
+        return format("config:edit org.opennms.plugins.cloud%n"
+                        + "property-set pas.tls.host %s%n"
+                        + "property-set pas.tls.port %s%n"
+                        + "property-set pas.tls.security TLS%n"
+                        + "property-set pas.mtls.host %s%n"
+                        + "property-set pas.mtls.port %s%n"
+                        + "property-set pas.mtls.security MTLS%n"
+                        + "property-set grpc.truststore \"%s\"%n"
+                        + "config:update",
+                "cloudMock",
+                9003,
+                "cloudMock",
+                9003,
+                classpathFileToString("/cert/horizon/servertruststore.pem")
+        );
+    }
+
+    public static String classpathFileToString(final String fileInClasspath) throws IOException {
+        URL url = new URL(format("jar:file:%s!%s", PATH_TO_CLOUD_MOCK, fileInClasspath));
+        try (InputStream inputStream = url.openStream()) {
+            return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        if (nonNull(stack)) {
```

Comment:
no need to stop explicitly. This is handled by the class rule. It may be event contra-productive as there is an option to keep the containers running intentionally for debugging purposes.

Reply:
ok